lightning-invoice/src/payment.rs

```diff
@@ -1613,20 +1597,6 @@ mod tests {
 		// Second path check
 		assert_eq!(inflight_map.0.get(&(3, false)).unwrap().clone(), 74);
 		assert_eq!(inflight_map.0.get(&(4, false)).unwrap().clone(), 64);
-
-		invoice_payer.handle_event(Event::PaymentPathSuccessful {
-			payment_id, payment_hash, path: route.paths[0].clone()
-		});
-
-		let inflight_map = invoice_payer.create_inflight_map();
-
-		assert_eq!(inflight_map.0.get(&(0, false)), None);
-		assert_eq!(inflight_map.0.get(&(1, true)), None);
-		assert_eq!(inflight_map.0.get(&(2, false)), None);
-
-		// Second path should still be inflight
-		assert_eq!(inflight_map.0.get(&(3, false)).unwrap().clone(), 74);
-		assert_eq!(inflight_map.0.get(&(4, false)).unwrap().clone(), 64)
```

Comment:
Doesn't look like there's much purpose for the test if it is no longer testing `InvoicePayer`. No sense testing `TestPayer`. Instead, I'd imagine we'd want similar test coverage for `ChannelManager` now.

Reply:
I think you're right -- essentially all this test currently does is to make sure `ChannelManager` generates `InflightHtlcs` with the correct values, which is already tested by `test_trivial_inflight_htlc_tracking`